src/lib/comms/include/sol-netctl.h

```diff
@@ -553,6 +696,81 @@ sol_netctl_find_service_by_name(const char *service_name)
 }
 
 /**
+ * @brief register a agent for network connection
+ *
+ * Register the agent for network connection.
+ * The register agent will inform the all information of network connection
+ * via the register callback struct.
+ * It includes the login input, the network connection failure and so on.
+ * The callback struct must be registered.
+ *
+ * @see struct sol_netctl_agent
+ *
+ * @param agent the agent Callback struct to be called when the related information is updated.
+ * @param data Add a user data per callback.
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int
+sol_netctl_register_agent(const struct sol_netctl_agent *agent, const void *data);
```

Comment:
you mean that the typedef of "sol_netctl_agent" replace "struct sol_netctl_agent".

If so, okay, sol_netctl_agent can be used.